Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM golang:1.18 as build-env
+
+WORKDIR /app
+
+COPY go.mod .
+
+RUN go mod download
+
+COPY . .
```

Comment:
it would be good to put all this starter for an image in a subdirectory, otherwise when this and PR#1 are merged this copy is going to grab everything in the repo